olp-cpp-sdk-core/src/http/curl/NetworkCurl.cpp

```diff
@@ -218,6 +219,39 @@ static curl_code SslctxFunction(CURL* curl, void* sslctx, void*) {
 #endif
 #endif
 
+struct curl_slist* GenerateRequestHeaders(const Headers& default_headers,
+                                          const Headers& request_headers,
+                                          const std::string& user_agent) {
+  struct curl_slist* headers = nullptr;
+
+  std::ostringstream sstrm;
+
+  for (const auto& header : default_headers) {
+    sstrm.str("");
+    sstrm << header.first;
+    sstrm << ": ";
+    sstrm << header.second;
```

Comment:
This seems to me a little overengineered. Why not use just a normal temp string with operator+:

```cpp
std::string curl_header = header.first + ": " + header.second. 
headers = curl_slist_append(headers, curl_header.c_str());
```